tests/testthat/test-code_dependency.R

```diff
@@ -0,0 +1,285 @@
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "b <- 2")
+
+  testthat::expect_identical(
+    get_code(q, names = "a"),
+    "a <- 1"
+  )
+  testthat::expect_identical(
+    get_code(q, names = "b"),
+    "b <- 2"
+  )
+})
+
+testthat::test_that("get_code extracts the code of a binding from a code put in an expression", {
+  q <- new_qenv()
+  q <- eval_code(q, expression(a <- 1))
+
+
+  testthat::expect_identical(
+    get_code(q, names = "a"),
+    "a <- 1"
+  )
+})
+
+testthat::test_that("get_code extracts the code of a binding from a code put in a language", {
+  q <- new_qenv()
+  q <- eval_code(q, quote(b <- 2))
+
+  testthat::expect_identical(
+    get_code(q, names = "b"),
+    "b <- 2"
+  )
+})
+
+testthat::test_that("get_code warns if binding doesn't exist in a code", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "b <- 2")
+
+  testthat::expect_warning(
+    get_code(q, names = "c")
+  )
+})
+
+
+testthat::test_that("get_code extracts code of a parent binding but only those evaluated before coocurence", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "b <- a")
+  q <- eval_code(q, "a <- 2")
+
+  testthat::expect_identical(
+    get_code(q, names = "b"),
+    c("a <- 1", "b <- a")
+  )
+})
+
+testthat::test_that("get_code extracts code of a parent binding if used in a function", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "b <- identity(x = a)")
+  q <- eval_code(q, "a <- 2")
+
+  testthat::expect_identical(
+    get_code(q, names = "b"),
+    c("a <- 1", "b <- identity(x = a)")
+  )
+})
+
+testthat::test_that("get_code is possible to output the code for multiple objects", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "b <- 2")
+  q <- eval_code(q, "c <- 3")
+
+  testthat::expect_identical(
+    get_code(q, names = c("a", "b")),
+    c("a <- 1", "b <- 2")
+  )
+})
+
+testthat::test_that("get_code can't extract the code when using assign", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "assign('b', 5)")
+  q <- eval_code(q, "b <- b + 2")
+  testthat::expect_identical(
+    get_code(q, names = "b"),
+    "b <- b + 2"
+  )
+})
+
+testthat::test_that("get_code can't extract the code when using assign, so use @effect tag", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "assign('b', 5) # @effect b")
+  q <- eval_code(q, "b <- b + 2")
+  testthat::expect_identical(
+    get_code(q, names = "b"),
+    c("assign(\"b\", 5)", "b <- b + 2")
+  )
+})
+
+testthat::test_that("get_code can't extract the code when using data", {
+  q <- new_qenv()
+  q <- eval_code(q, "data(iris)")
+  q <- eval_code(q, "iris2 <- head(iris)")
+  testthat::expect_identical(
+    get_code(q, names = "iris2"),
+    "iris2 <- head(iris)"
+  )
+})
+
+testthat::test_that("get_code can't extract the code when using data, so use @effect tag", {
```

Comment:
I suggest following. I hope "rhs" I understandable by all of us - if not, please propose different description. 
```suggestion
testthat::test_that("get_code can't extract the code when function creates an object which is used only on rhs", {
```

Reply:
I think `rhs` is common enough to be assumed to be understood. However, see my general comment about tags being mandatory.

Reply:
if we want to tag every line, it's easier just to brake the code for each object separately, as it is currently